CRM/Core/Payment/PayPalIPN.php

```diff
@@ -379,7 +380,11 @@ public function main() {
 
       $status = $input['paymentStatus'];
       if ($status === 'Denied' || $status === 'Failed' || $status === 'Voided') {
-        $this->failed($objects);
+        Contribution::update(FALSE)->setValues([
+          'cancel_date' => 'now',
+          'contribution_status_id:name' => 'Cancelled',
+        ])->addWhere('id', '=', $contributionID)->execute();
+        Civi::log()->debug("Setting contribution status to Failed");
```

Comment:
@eileenmcnaughton The comment is not correct anymore - we are actually setting it to "Cancelled", unless that's the typo?